packages/graphql/src/services/sessions.ts

```diff
@@ -1638,11 +1623,15 @@ export async function getPinnedFeedbacks(
   return reducedSession
 }
 
-function checkCorrectnessFreeText(instance) {
+function checkCorrectnessFreeText(instance: AllQuestionInstanceTypeData) {
   // Adds "correct" attribute (true/false) to results in FREE_TEXT questions if they match any given solution)(exact match, case insensitive)
-  if (instance.questionData.type === 'FREE_TEXT') {
+  instance.elementType = instance.questionData.type
+  if (
+    instance.elementType === 'FREE_TEXT' &&
```

Comment:
this is a very ugly trial to get results and questionData type narrowing to work within the same function. we need to find a better way for this, maybe add the elementType to the data schema of the instance in addition to the type in questionData, and use instance.elementType for all checks, also to detect the type of questionData.